### PR TITLE
feat: Add option for embedding attachments in definitions

### PR DESCRIPTION
**Pull request by hofmeister**
Needed for adding additional config files into the YML

### CODE REVIEW

File: packages/go/schemas/types/core/attachment.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "value"
+      ],
+      "properties": {
+        "type": {
+          "title": "AttachmentContentType",
+          "type": "string",
+          "enum": [
+            "url",
+            "base64",
+            "base64-gzip",
```
Comment (sorenmat) [inline packages/go/schemas/types/core/attachment.json:36]: perhaps base64+gzip isn't that the style you use in http content types?

File: packages/go/schemas/types/core/attachment.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  "additionalProperties": false,
+  "required": [
+    "filename",
+    "content"
+  ],
+  "properties": {
+    "filename": {
+      "type": "string"
+    },
+    "contentType": {
```
Comment (sorenmat) [inline packages/go/schemas/types/core/attachment.json:16]: what's the idea of having both this and AttachmentContentType?

File: packages/go/schemas/types/core/attachment.json (hunk 1 above)
Comment (hofmeister) [reply packages/go/schemas/types/core/attachment.json:36]: This isn't a http content type though ?

File: packages/go/schemas/types/core/attachment.json (hunk 2 above)
Comment (hofmeister) [reply packages/go/schemas/types/core/attachment.json:16]: I'll rename to "AttachmentContentFormat" - is more accurate

File: packages/go/schemas/types/core/attachment.json (hunk 1 above)
Comment (sorenmat) [reply packages/go/schemas/types/core/attachment.json:36]: I know, both just for making it look familiar

**Review by morten-holm — APPROVED** [review]

File: packages/go/schemas/types/core/attachment.json (hunk 1 above)
Comment (hofmeister) [reply packages/go/schemas/types/core/attachment.json:36]: Renamed to format - you're right this was confusing

**Review by sorenmat — APPROVED** [review]
Other then that, looks GTM

File: packages/go/model/model.go [hunk 3]
```diff
@@ -1,9 +1,23 @@
 package model
 
 type BlockDefinition struct {
-	Kind     string              `json:"kind"`    
-	Metadata Metadata            `json:"metadata"`
-	Spec     BlockDefinitionSpec `json:"spec"`    
+	Attachments []Attachment        `json:"attachments,omitempty"`
```
[the remark this answers is not in the document]
Comment (hofmeister) [reply packages/go/model/model.go:4]: Codegen thing doesnt really do that for anything. But you're right

**Pull request merged**